src/Discovery/index.tsx

```diff
@@ -99,7 +99,8 @@ const DiscoveryWithMDSBackend: React.FC<{
           : numberOfStudiesForAllStudiesBatch);
       } else {
         loadStudiesFunction = loadStudiesFromMDS;
-        loadStudiesParameters = [props.config?.features?.guidType, 10, false];
+        loadStudiesParameters = (numberOfBatchesLoaded === 1
+          ? [props.config?.features?.guidType, 10, false] : [props.config?.features?.guidType, 2000, true]);
```

Comment:
Correct me if I misunderstood. In line 80, we have `expectedNumberOfTotalBatches` set to 2, and here we have a ternary which checks if `numberOfBatchesLoaded` === 1, but since we have line 89, which doesn't let us ever reach `numberOfBatchesLoaded` as `1`, the true condition never works. Right? Excuse me if i'm hugely mistaken 😅

Reply:
`numberOfBatchesLoaded` will reach 1 because of L89, but you are right, this logic can be improved to save an unncecssary loop cycle. I'll update